src/main/java/uk/gov/hmcts/reform/pip/rules/repository/InMemoryRepository.java

```diff
@@ -71,5 +106,39 @@
     }
 
 
+    /**
+     * Method to retrieve a court hearings using court ID.
+     * @param courtId The ID of the court.
+     * @return The court hearings to return, if found.
+     */
+    public Optional<CourtHearings> getCourtHearings(Integer courtId)
+    {
+        CourtHearings court = this.courtHearings.stream()
+            .filter(c -> courtId.equals(c.getCourtId()))
+            .findAny()
+            .orElse(null);
+
+        List<Hearing> listHearings = this.listHearings.stream()
+            .filter(h -> h.getCourtId() == courtId)
```

Comment:
most of this method seems to be attaching the hearings to the court, can this be split out

Reply:
done